src/lib.rs

```diff
@@ -1264,6 +1277,61 @@ where
     }
 }
 
+/// A streaming iterator that flattens nested streaming iterators.
+#[derive(Debug)]
+pub struct Flatten<I, J> {
+    iter: I,
+    _inner: PhantomData<J>,
+}
+
+impl<I, J> StreamingIterator for Flatten<I, J>
+where
+    I: StreamingIteratorMut<Item = J>,
+    J: StreamingIterator,
+{
+    type Item = J::Item;
+
+    #[inline]
+    fn advance(&mut self) {
+        loop {
+            if let Some(ref mut iter) = self.iter.get_mut() {
```

Comment:
`get_mut` documents:
> The behavior of calling this method before `advance` has been called is unspecified.

I think we'll need a `first: bool` flag for the first round, and then we should also not call `advance` any more after seeing `get_mut() -> None` or `is_done()`.